src/NaluParsing.C

```diff
@@ -1144,7 +1144,13 @@ namespace YAML
   bool convert<sierra::nalu::SymmetryUserData>::decode(const Node& node,
     sierra::nalu::SymmetryUserData& symmetryData)
   {
-    // nothing as of yet
+    // This allows the user to set a fixed temperature gradient that is
+    // achieved through application of a compatible heat flux. 
+    if (node["temperature_gradient"])
+    {
+      symmetryData.temperatureGradient_ = node["temperature_gradient"].as<sierra::nalu::NormalHeatFlux>();
```

Comment:
As the intent of the bc is to extract a double, we can simply use the double type here as apposed to a new NormalHeatFlux() type. Either way is fine; this is very clear. However, if we had a special type for every double, the parsing code would substantially grow. Your choice as to whether or not the type safety is required.

Reply:
@spdomin Considering that the first thing that is done is extracting the 1 double out of the special struct and passed around as double it doesn't make sense to have a special data type for the user data. Same can be said of roughnessHeight etc.

Reply:
@sayerhs, I agree; let's have him change this to a double - along with the other new parsing type. This will make things simpler in the end and reduce code.